### DISCUSSION TITLE
What is an active tag?

### DISCUSSION THREAD

**Discussion by emiham** [ad-si/TaskLite#74] [Q&A] [answered]
The documentation says a project in TaskLite is an "active tag". It doesn't however mention what an active tag is. I have tasks with tags, but none of them show up as projects. How do I change this?

**Comment by ad-si** [comment, marked as answer]
It just means that that not all issues with this particular tag are exclusively open or exclusively closed, aka you're actively working on issues with this tag.
Does this make sense?

I was also thinking about introducing a way to mark tags as projects. That might be a more precise way, but requires additional work by users.

**Comment by emiham** [comment]
Thank you, I finally understand how it works! 

To me I don't think the current `projects` command is very useful, if it's basically just `tags` with a filter. I think a way of marking tags as projects would be good, or perhaps some sort of notion of primary and secondary tags to not force users into a certain workflow.

**Reply by ad-si** [reply]
Another idea that's becoming more frequent is `key:value` tags. (E.g. GitLab's [Scoped labels](https://docs.gitlab.com/ee/user/project/labels.html#scoped-labels))
That might be a nice solution as you can then have a `project:myapp` tag.

**Reply by ad-si** [reply]
But TaskLite actually kind of already supports this with the metadata field. So I guess the KISS way of doing this would be to write yourself a simple shell script as seen here: https://tasklite.org/cli/usage#metadata
(Or an External Command as seen here: https://tasklite.org/cli/usage#external-commands)

**Reply by emiham** [reply]
Thanks for all the suggestions! After thinking about it today I think TaskLite might not be the right tool for me. I really like how powerful it is, allowing you to be very flexible, but at the end of the day I don't need that many things from my to do list, so it ends up being a bit too much to implement myself I think.

**Reply by ad-si** [reply]
Yeah, fair enough 👍
The name is also quite misleading by now. There is nothing "Lite" about it. 😅 (The name refers to SQLite, but that's not really clear. Maybe I should rename it)
It'd be interested what you'll pick instead!

**Reply by emiham** [reply]
To me at least I feel like the documentation sells it as somewhat minimalistic. 
> TaskLite tries to adhere to the Unix philosophy: Do one thing and do it well. This, however, also means that it intentionally does not include certain features, which you might expect it to have.

> **Simpler**
> Taskwarrior has several redundant features and unnecessarily re-implements shell features like aliases.

I don't think these statements are false (although bundling a web server feels like it goes against the first one a bit), but I think they mislead me alongside the name into getting the wrong impression of the program. I honestly kind of wish I could think of a way to use SQL queries in my workflow right now as it sounds super cool, but it might just be that I don't need it. 

>  It'd be interested what you'll pick instead!

I've settled on todo.txt for the last week or so. There has been a fair bit of work I've had to do by myself, but it felt better building on top of a very minimal tool rather than building parallel to features I wouldn't be using. As an aside, I was using `tu` for a bit and really enjoyed it, I only stopped because I started using `topydo` for recurrence and scheduling so it wasn't really needed anymore.

**Reply by gour** [reply]
> I've settled on todo.txt for the last week or so. There has been a fair bit of work I've had to do by myself, but it felt better building on top of a very minimal tool rather than building parallel to features I wouldn't be using. As an aside, I was using `tu` for a bit and really enjoyed it, I only stopped because I started using `topydo` for recurrence and scheduling so it wasn't really needed anymore.

If you're into Python, I might suggest you to have a look at [Etm](https://github.com/dagraham/etm-dgraham).
